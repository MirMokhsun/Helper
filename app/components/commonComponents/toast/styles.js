import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    background: {
        flex: 1,
        width: '100%',
        height: '100%',
        justifyContent: "center",
        alignItems: 'center',
        backgroundColor: 'transparent',
        position: 'absolute',
        zIndex: 1000000,
        padding: 20,
    },
});