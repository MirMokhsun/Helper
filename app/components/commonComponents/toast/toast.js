import React from 'react';
import { Text, View } from 'react-native';
import { styles } from './styles';
import store from '../../../redux/store/store';

export class Toast extends React.Component {
    constructor(props) {
        super(props);
        this.hideToast = props.hideToast
    }

    componentDidUpdate(prevProps) {
        const { isVisibleToast } = this.props;
        if (prevProps.isVisibleToast === false && isVisibleToast === true) {
            setTimeout(() => {
                store.dispatch(this.hideToast());
            }, 3000);
        }

    }

    render() {
        const { toastText, isVisibleToast } = this.props;
        const toast = isVisibleToast ? (
            <View style={styles.background} pointerEvents="none">
                <View style={{ backgroundColor: '#666', borderRadius: 30, padding: 15 }}>
                    <Text style={{ color: '#FFFFFF', textAlign: 'center' }}>{toastText}</Text>
                </View>
            </View>
        ) : null;
        return toast;
    }
}
