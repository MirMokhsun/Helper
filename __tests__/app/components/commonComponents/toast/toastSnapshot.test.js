import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Toast } from "../../../../../app/components/commonComponents/toast/toast";

it('create snapshot Toast', () => {
    const renderer = new ShallowRenderer()
    let toast = renderer.render(<Toast />);
    expect(toast).toMatchSnapshot();
});

it('create snapshot Toast', () => {
    const renderer = new ShallowRenderer()
    let toast = renderer.render(<Toast {...{toastText: 'bla bla bla', isVisibleToast: 'true'}} />);
    expect(toast).toMatchSnapshot();
});
