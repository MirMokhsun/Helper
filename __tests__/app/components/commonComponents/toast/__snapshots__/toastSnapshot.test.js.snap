// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot Toast 1`] = `null`;

exports[`create snapshot Toast 2`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "padding": 20,
      "position": "absolute",
      "width": "100%",
      "zIndex": 1000000,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#666",
        "borderRadius": 30,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "textAlign": "center",
        }
      }
    >
      bla bla bla
    </Text>
  </View>
</View>
`;
